src/pipelines/rust_app.rs

```diff
@@ -57,6 +61,7 @@ impl RustApp {
             })
             .unwrap_or_else(|| html_dir.join("Cargo.toml"));
         let bin = el.attr("data-bin").map(|val| val.to_string());
+        let wasm_opt = el.attr("wasm-opt").map(|val| val.to_string());
```

Comment:
Let's add a struct to wrap and validate this value. Something like (using https://jeltef.github.io/derive_more/derive_more/deref.html):

```rust
#[derive(Deref)]
#[deref(forward)]
struct WasmOptValue(String);

impl WasmOptValue {
    pub fn validate(val: &str) -> Result<Self> {
        // validate the value then wrap it in self
    }
}
```

Then we can update this line to be `let wasm_opt = el.attr("wasm-opt").map(|val| WasmOptValue::validate(val)).transpose()?;`.

Then, down in the code where it is actually used, we can simply do `self.wasm_opt.as_deref().as_str()` which is compatible with the `AsRef<std::ffi::OsStr>` constraint used below.

This will allow folks to see if they've supplied an incorrect value as soon as Trunk is invoked, instead of having to wait until the very end of the cargo build.

Reply:
How about creating an enum for the possible values and implement `FromStr` and `AsRef<str>` on it?
That would be a more type safe approach.

Initially I thought it's okay to let the user just pass any value as `wasm-opt` would just throw an error and we wouldn't have to adjust whenever `wasm-opt` changes its options. But especially this option probably never changes (or extremely rarely) and can be adjusted quickly.

Reply:
I dig it. I was going to suggest that initially, but I wasn't sure if it would work with structopt. But if it does, then I definitely agree that it is a better approach.

Reply:
Just checked, `FromStr` is what structopt requires for the argument types.

https://docs.rs/structopt/0.3.21/structopt/#type-magic

So this should work just fine with enum and `FromStr` 👍